yew/src/virtual_dom/vtag.rs

```diff
@@ -249,36 +244,81 @@ impl VTag {
         }
     }
 
-    /// This handles patching of attributes when the keys are equal but
-    /// the values are different.
-    fn diff_attributes<'a>(
-        &'a self,
-        ancestor: &'a Option<Box<Self>>,
-    ) -> impl Iterator<Item = Patch<&'a str, &'a str>> + 'a {
-        // Only change what is necessary.
-        let to_add_or_replace =
-            self.attributes.iter().filter_map(move |(key, value)| {
-                match ancestor
-                    .as_ref()
-                    .and_then(|ancestor| ancestor.attributes.get(&**key))
-                {
-                    None => Some(Patch::Add(&**key, &**value)),
-                    Some(ancestor_value) if value != ancestor_value => {
-                        Some(Patch::Replace(&**key, &**value))
-                    }
-                    _ => None,
+    /// Diffs and patches changes in attributes between the old and new list.
```

Comment:
The point is that there is a need for allocating the patches. By computing all necessary patches upfront, they can be applied all at once rather than interleaved with diff computation.

Reply:
Okay, so you are saying in the original implementation the iterator was inlined inside the consuming for loop? But that still adds the overhead of packing the changes into an iterator of `Patch` items and then immediately consuming it with the added overhead of a match expression. I can produce a benchmark for this as well, if need be.

Reply:
> Okay, so you are saying in the original implementation the iterator was inlined inside the consuming for loop?

I'm not talking about the original implementation. If we were talking about the original implementation your change is an improvement because it reduces allocations.

I'm talking about the future implementation. Yew needs to batch DOM writes together at a high level. Imagine you have hundreds of DOM patches across many elements and components from one click of a button. Those should be computed first, and then all patches should be applied together in the rendering phase. Does that make sense?

> I can produce a benchmark for this as well, if need be

I don't think a benchmark is needed, this is a macro level optimization.

Reply:
Oh, you meant batching between multiple VTags. Okay, that makes sense. Will revert.

Reply:
Ah yes, sorry for not being clearer. Thanks!